### ISSUE TITLE
Unhandled Exception: FileSystemException: Cannot open file

### ISSUE THREAD

**Issue by KennyTC** [faithoflifedev/google_vision_workspace#3]
I use the credentials json file (which I downloaded when creating Service Account). It has the following format
{
  "type": "service_account",
  "project_id": "",
  "private_key_id": "",
  "private_key": "",
  "client_email": "",
  "client_id": "",
  "auth_uri": "",
  "token_uri": "",
  "auth_provider_x509_cert_url": "",
  "client_x509_cert_url": ""
}
I put the file in "assets/a.json" and use the following code to read.

```
final String response = await rootBundle.loadString('assets/a.json');
final data = await json.decode(response);

final googleVision = await GoogleVision.withJwt("assets/a.json");
print(googleVision.toString());
```
then, the rootBundle can read the json file well, but GoogleVision.withJwt has error "Unhandled Exception: FileSystemException: Cannot open file"

**Comment by faithoflifedev** [comment]
Hi @KennyTC, based on the sample code you've provided, you should be doing the following:

```
Future<File> getFileFromAsset(String path) async {
    final byteData = await rootBundle.load('assets/$path');
    
    final buffer = byteData.buffer;
    
    final tempDir = await getTemporaryDirectory();
    
    final filePath = tempDir.path+ '/file_01.tmp'; // file_01.tmp is dump file, can be anything
 
    return File(filePath).writeAsBytes(buffer.asUint8List(byteData.offsetInBytes, byteData.lengthInBytes));
}

final googleVision = await GoogleVision.withJwt(getFileFromAsset('assets/a.json'));
```

I'll update the code in a future release so that the more obvious method (that you used) is also an option.  My initial use case for this package was desktop, so not a lot of thought was put into mobile.

**Comment by KennyTC** [comment]
withJwt accepts  input as a string, but  getFileFromAsset returns Future<File>, so the error "The argument type 'Future<File>' can't be assigned to the parameter type 'String'"

**Comment by faithoflifedev** [comment]
Sorry @KennyTC , I didn't test the code on a real project and I missed a few things.

```
Future<String> getFileFromAsset(String path) async {
    final byteData = await rootBundle.load('assets/$path');
    
    final buffer = byteData.buffer;
    
    final tempDir = await getTemporaryDirectory();
    
    return tempDir.path + '/file_01.tmp'; // file_01.tmp is dump file, can be anything
}

// returns a string with the temp file name created from the asset
final jwtFromAsset = await getFileFromAsset('assets/a.json')

final googleVision = await GoogleVision.withJwt(jwtFromAsset);
```

Hopefully that will do the trick (I still don't have access to my dev machine) :-).

**Comment by KennyTC** [comment]
I am pretty  new with  Flutter,  I  am  sorry if my question is  obvious to you.

- Is getTemporaryDirectory() from `path_provider` library?
- `final jwtFromAsset = await getFileFromAsset('assets/a.json')` -->should  be   `final jwtFromAsset = await getFileFromAsset('a.json')` because  you also used  `assets` in `final byteData = await rootBundle.load('assets/$path');`?
- After fixing the above dependencies, the  program gets error `/data/user/0/com.example.example2/cache/file_01.tmp' (OS Error: No such file or directory, errno = 2)`,  (  the  variable `tempDir` is  `/data/user/0/com.example.example2/cache` in   this  case).   I   do  not   know  where `/data/user/0/com.example.example2/cache` is.

**Comment by faithoflifedev** [comment]
@KennyTC, I think I have it correct this time.

```
Future<String> getFileFromAsset(String path) async {
    final byteData = await rootBundle.load('assets/$path');
    
    final buffer = byteData.buffer;
    
    final tempDir = await getTemporaryDirectory();

    final filePath = tempDir.path+ '/file_01.tmp'; // file_01.tmp is dump file, can be anything

    File(filePath).writeAsBytes(buffer.asUint8List(byteData.offsetInBytes, byteData.lengthInBytes));
    
    return filePath ;
}

// returns a string with the temp file name created from the asset
final jwtFromAsset = await getFileFromAsset('a.json')

final googleVision = await GoogleVision.withJwt(jwtFromAsset);
```

In the previous edit I removed the code that actually writes the file to the temp dir.

**Comment by KennyTC** [comment]
Thanks very  much.  I am really  appreciate. 
I used your newest version, then I get error at the  
`final googleVision = await GoogleVision.withJwt(jwtFromAsset);`. The  error is as follows:

![image](https://user-images.githubusercontent.com/16069257/210656022-0ccd5b83-a45d-4b0b-b50d-423531180666.png)

It seems my json is not the right format?

**Comment by bigjjjim** [comment]
Hi KennyTC, 

I was having the same troubles. It could be clearer that the service.json has to be loaded from directory and not from Flutter assets... 

This code to save the file and get the path worked for me: 

```  
Future<String> _getImageFileFromAssets(String path) async {
    Directory tempDir = await getTemporaryDirectory();
    String tempPath = tempDir.path;
    var filePath = "$tempPath/$path";
    var file = File(filePath);
    // if (file.existsSync()) {
    //   return file.path;
    // } else {
      final byteData = await rootBundle.load('assets/$path');
      final buffer = byteData.buffer;
      await file.create(recursive: true);
       await file
          .writeAsBytes(buffer.asUint8List(byteData.offsetInBytes,
          byteData.lengthInBytes));
          return file.path;
    // }
   
  }
```

**Comment by faithoflifedev** [comment]
@KennyTC,

Finally have my dev machine again.

This also works:

```
Future<String> getFileFromAsset(String path) async {
    final byteData = await rootBundle.load('assets/$path');
    
    final buffer = byteData.buffer;
    
    final tempDir = await getTemporaryDirectory();

    final filePath = tempDir.path+ '/file_01.tmp'; // file_01.tmp is dump file, can be anything

    await File(filePath).writeAsBytes(buffer.asUint8List(byteData.offsetInBytes, byteData.lengthInBytes));
    
    return filePath ;
}
```

Notice the addition of the `await` on the line where the temp file is created.

I'm working on rehashing my command line example to work with flutter, hopefully I'll be able to publish the changes in the next day or so.

**Comment by faithoflifedev** [comment]
Hi @KennyTC and @bigjjjim ,

I just published [v1.0.5+4](https://pub.dev/packages/google_vision/versions/1.0.5+4) which includes a Flutter example based on the above code.
